### PR TITLE
fix(1393): Add functional test for triggers

### PR DESCRIPTION
**Pull request by tkyi**
## Context

Missing trigger functional tests.

## Objective

This PR adds some triggers functional tests. Also updates to comply with deprecation: https://github.com/cucumber/cucumber-js/blob/master/CHANGELOG.md#deprecations

Issues:
- ~new parallel fork join feature breaks rule: external parallel triggered builds should be a part of the same event if triggered by the same build~ _fixed in https://github.com/screwdriver-cd/screwdriver/pull/1986_
- cannot delete and recreate pipeline as the "requires" triggers need to be hardcoded in the screwdriver.yaml

TODO:
- refactor to clean up code
- add parallel fork join tests?

## References

Related to https://github.com/screwdriver-cd/screwdriver/issues/1393

## License
I confirm that this contribution is made under a BSD license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: features/step_definitions/trigger.js [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+        shouldNotDeletePipeline: true });
+});
+
+When(/^the "(.*)" job on branch "(?:.*)" is started$/,
+    { timeout: TIMEOUT }, function step(jobName) {
+        let jobId;
+        let buildVarName;
+
+        switch (jobName) {
+        case 'success_A':
```
Comment (klu909) [inline features/step_definitions/trigger.js]: can we use hashmaps here instead so we don't need to have long switch like this

Comment (klu909) [reply features/step_definitions/trigger.js]: one hashmap to map`fail_A` to `successAJobId` and `successABuildId`?

Comment (tkyi) [reply features/step_definitions/trigger.js]: updated

**Comment by tkyi** [comment]
@jithine https://github.com/screwdriver-cd-test/functional-trigger/blob/pipelineB/screwdriver.yaml

**Pull request merged**